File: lib/data/models/exercise.dart
class Exercise {
  final String id;
  final String name;
  final String muscle;
  final String equipment;
  final String difficulty;
  final String instructions;
  final int sets;
  final String reps;
  final int restSeconds;

  const Exercise({
    required this.id,
    required this.name,
    required this.muscle,
    required this.equipment,
    required this.difficulty,
    required this.instructions,
    required this.sets,
    required this.reps,
    required this.restSeconds,
  });
}

File: backend_laravel_starter/app/Models/Exercise.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class Exercise extends Model
{
    protected $fillable = [
        'name',
        'slug',
        'description',
        'primary_muscle',
        'equipment',
        'difficulty',
        'instructions',
        'sets',
        'reps',
        'rest_seconds',
        'media_url',
        'thumbnail_url',
        'is_active',
    ];

    protected $casts = [
        'is_active' => 'boolean',
    ];
}
